### ISSUE TITLE
Feature Request: default/auto parameters for training

### ISSUE THREAD

**Issue by chreul** [Calamari-OCR/calamari#33]
when training a model ensemble it would be nice if there was an "auto" option to automatically choose somewhat sane parameters, e.g. regarding early stopping. these parameters should depend on the number of GT lines available for training.
based on our experiments i propose the following auto defaults:
- early_stopping_frequency: about half the number of available GT lines, maybe rounded up to the next hundred.
- early_stopping_nbest: 5, i think 10 is too high. this is a general "issue" and not specific to this auto functionality.
- max_iters: maybe 10 epochs? not sure about this one.
- checkpoint_frequency = early_stopping_frequency.

**Comment by ChWick** [comment]
* As of bb79ecf: early stopping frequencies may be relative to epochs (if <= 1), default = 0.5.
* early stopping number of best models set to 5
* max iterations is unchanged, usually this parameter should never be required, because training with a validation set is recommended. Alternatively, the training can always be stopped manually.
* By default the checkpoint frequency uses the early stopping frequency if not explicitly stated.
